src/pretix/presale/checkoutflow.py

```diff
@@ -700,6 +702,18 @@ def post(self, request, *args, **kwargs):
                        sales_channel=request.sales_channel.identifier)
 
 
+def business_heuristic(organizer):
+    cached_result = organizer.cache.get('checkout_heuristic_is_business')
+    if cached_result is None:
+        result = InvoiceAddress.objects.filter(order__event__organizer=organizer).aggregate(
+            total=Count('*'), business=Sum(Cast('is_business', output_field=models.IntegerField())))
+        is_business = result['business'] / result['total'] >= 0.6
+        organizer.cache.set('checkout_heuristic_is_business', str(is_business), timeout=15)
```

Comment:
String is because the cache API seems to only accept strings.

The 15s was from testing and then forgotten, I'll change it to ... maybe an hour?

Reply:
Ah, sorry, it is not django cache, but a custom one for isolation.

An hour sound reasonable. But @raphaelm probably has input on it as well. :)

Reply:
I think the cache would accept booleans :thinking: But not worth diving too deep into it for this.

For cache timeout, see other conversation.

Reply:
The custom cache has a type annotation that says otherwise, https://github.com/pretix/pretix/blob/master/src/pretix/base/cache.py#L65 

However this annotation seems to be incorrect, as it relies on Django cache internally, which pickles everything, so bool should be okay,